### PR TITLE
add `GlobVec`, `always_copy_files` and `always_include_files`

### PR DESCRIPTION
**Pull request by wolfv**
cc @orhun you also did some work on glob's recently.
This introduces a new type that renders them as regular strings, but also eagerly parses them so that we can emit errors right away.

### CODE REVIEW

File: src/recipe/parser/glob_vec.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        }
+    }
+}
+
+impl GlobVec {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn globset(&self) -> Option<&GlobSet> {
```
Comment (orhun) [inline src/recipe/parser/glob_vec.rs:54]: Should we also add a getter method for getting the underlying Vector of `GlobMatcher`s? Or is `GlobSet` enough to perform matches?

**Review by orhun — COMMENTED** [review]
I like this! Just had one comment since I mainly used `GlobMatcher`s.

File: src/recipe/parser/glob_vec.rs (hunk 1 above)
Comment (wolfv) [reply src/recipe/parser/glob_vec.rs:54]: Let's do that when we need it? Maybe `globset` is enough -- that depends I'd say.

Comment (orhun) [reply src/recipe/parser/glob_vec.rs:54]: Alright! I will switch to using `GlobVec` after this PR is merged and see if that's necessary.

**Review by orhun — APPROVED** [review]

**Pull request merged**